Tests/NIOCoreTests/ByteBufferTest.swift

```diff
@@ -1754,14 +1754,24 @@ class ByteBufferTest: XCTestCase {
             throw XCTSkip("This test is only supported on 64-bit systems.")
         }
 
+        // This allocator assumes that we'll never call realloc.
+        let fakeAllocator = ByteBufferAllocator(
+            hookedMalloc: { _ in .init(bitPattern: 0xdeadbeef) },
+            hookedRealloc: { _, _ in fatalError() },
+            hookedFree: { precondition($0 == .init(bitPattern: 0xdeadbeef)!) },
+            hookedMemcpy: {_, _, _ in }
+        )
+
         let targetSize = Int(UInt32.max)
-        var buffer = self.allocator.buffer(capacity: targetSize)
+        var buffer = fakeAllocator.buffer(capacity: targetSize)
 
         // Move the reader index forward such that we hit the slow path.
         let offset = Int(_UInt24.max) + 1
         buffer.moveWriterIndex(to: offset)
         buffer.moveReaderIndex(to: offset)
-        buffer.writeInteger(UInt32(0))
```

Comment:
@Lukasa you can still make this write happen by having the fake allocator do

```
let ptr = malloc(4)
return ptr - offset
```

And `free` would do `free(ptr + offset)`.

Reply:
Sure, but the write isn't important: we never validated the bytes came back.